### PR TITLE
Hive monad

### LINKED ISSUE #250: Create Hive table with text format from Thrift struct

### PR DESCRIPTION
**Pull request by stephanh**
Takes advantage of the Hive monad from Ebenezer and all corresponding changes.

* `Result` is now in omnitool instead of permafrost.
* Use `fold` on `Result` instead of pattern matching.
* `fromHive` for Execution.
* Deprecated `QueryExecution`.
* Added Hive and Hdfs to api package.

This also closes #250.

### CODE REVIEW

File: maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/exec/QueryExecution.scala [hunk 1]
```diff
@@ -31,6 +31,7 @@ import au.com.cba.omnia.maestro.core.hive.HiveTable
   * @param queries:  The hive queries to run
   * @param settings: Hive settings. Empty by default.
   */
+@deprecated("Use the query/queries method in Hive instead", "1.15")
```
Comment (SamRoberts) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/exec/QueryExecution.scala]: :+1: for making use of deprecated tags. Should we ensure we have a github issue to remove these methods at a later date?

Comment (stephanh) [inline maestro-core/src/main/scala/au/com/cba/omnia/maestro/core/exec/QueryExecution.scala]: Added an issue.

File: maestro-example/src/main/scala/au/com/cba/omnia/maestro/example/CustomerUploadExample.scala [hunk 2]
```diff
@@ -35,15 +33,21 @@ class CustomerUploadExample(args: Args) extends Maestro[Customer](args) {
...
-    case Error(_) => {
+  List(byDateResult, byIdResult).sequence_.foldAll(
+    result => {
+      // continue with post-processing here
+    },
+    errorMsg => {
+      // do not post-process files when there was an error copying them
+      // report the error here
+    },
+    execption => {
```
Comment (SamRoberts) [inline maestro-example/src/main/scala/au/com/cba/omnia/maestro/example/CustomerUploadExample.scala]: execption -> exception

**Comment by SamRoberts** [comment]
:+1:

File: maestro-example/src/main/scala/au/com/cba/omnia/maestro/example/CustomerUploadExample.scala (hunk 2 above)
Comment (stephanh) [inline maestro-example/src/main/scala/au/com/cba/omnia/maestro/example/CustomerUploadExample.scala]: Fixed

**Pull request merged**